// vanilla-calendar/build/js/component/grid/Day.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Day = /** @class */ (function () {
    function Day(parent, reference, date, today, daySelector) {
        var _this = this;
        this.onClick = function (e) {
            _this.daySelector
                .setDate(_this.date)
                .appendTo(_this.parent)
                .setPositionLeft((100 / 7) * _this.date.getDay() + "%")
                .setWidth(100 / 7 + "%")
                .setDisplayBlock();
        };
        this.daySelector = daySelector;
        this.today = today;
        this.date = date;
        this.element = document.createElement('div');
        this.parent = parent;
        this.updateUI(reference);
        this.parent.append(this.element);
        this.setClickEvent();
    }
    Day.prototype.setDate = function (date) {
        this.date = date;
    };
    Day.prototype.getDate = function () {
        return this.date;
    };
    Day.prototype.updateUI = function (reference) {
        this.element.className = 'day';
        if (this.date.getDay() === 0) {
            this.element.className = 'sunday day';
        }
        if (this.date.getMonth() !== reference.getMonth()) {
            if (this.date.getDay() === 0) {
                this.element.className = 'day sunday-not-this-month';
            }
            else {
                this.element.className = 'day day-not-this-month';
            }
        }
        if (this.date.getFullYear() === this.today.getFullYear() &&
            this.date.getDate() === this.today.getDate() &&
            this.date.getMonth() === this.today.getMonth()) {
            this.element.innerHTML = "\n        <div class=\"day-today-container\">\n          <span>" + this.date.getDate() + "</span>\n          <span class=\"today-text\">today</span>\n        </div>";
        }
        else {
            this.element.innerHTML = "" + this.date.getDate();
        }
    };
    Day.prototype.setClickEvent = function () {
        this.element.addEventListener('click', this.onClick);
    };
    return Day;
}());
exports.default = Day;
