// vanilla-calendar/build/js/component/grid/DaySelector.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var DaySelector = /** @class */ (function () {
    function DaySelector() {
        this.element = document.createElement('div');
        this.element.className = 'day-selector';
    }
    DaySelector.prototype.setDate = function (date) {
        this.date = date;
        return this;
    };
    DaySelector.prototype.setPositionLeft = function (positionLeft) {
        this.element.style.left = "" + positionLeft;
        return this;
    };
    DaySelector.prototype.appendTo = function (parent) {
        parent.append(this.element);
        return this;
    };
    DaySelector.prototype.setWidth = function (width) {
        this.element.style.width = width;
        return this;
    };
    DaySelector.prototype.setDisplayNone = function () {
        this.element.style.display = 'none';
        return this;
    };
    DaySelector.prototype.setDisplayBlock = function () {
        this.element.style.display = 'block';
        return this;
    };
    return DaySelector;
}());
exports.default = DaySelector;
